fe/src/main/java/org/apache/doris/catalog/RangePartitionInfo.java

```diff
@@ -93,12 +93,60 @@ public void addPartition(long partitionId, Range<PartitionKey> range, DataProper
     }
 
     public Range<PartitionKey> checkAndCreateRange(SingleRangePartitionDesc desc) throws DdlException {
+        List<Map.Entry<Long, Range<PartitionKey>>> entries =
+                new ArrayList<Map.Entry<Long, Range<PartitionKey>>>(this.idToRange.entrySet());
+        Collections.sort(entries, RANGE_MAP_ENTRY_COMPARATOR);
+        PartitionKeyDesc partKeyDesc = desc.getPartitionKeyDesc();
+        if (partKeyDesc.getLowerValues().isEmpty()) {
+            return handleLessThanRange(desc, entries);
+        } else {
+            return handleFixedRange(desc, entries);
+        }
+    }
+
+    private Range<PartitionKey> handleFixedRange(SingleRangePartitionDesc desc, List<Map.Entry<Long, Range<PartitionKey>>> entries) throws DdlException {
+        Range<PartitionKey> newRange = null;
+        PartitionKeyDesc partKeyDesc = desc.getPartitionKeyDesc();
+        try {
+            PartitionKey lowKey =
+                    PartitionKey.createPartitionKey(partKeyDesc.getLowerValues(), partitionColumns);
+            PartitionKey upperKey =
+                    PartitionKey.createPartitionKey(partKeyDesc.getUpperValues(), partitionColumns);
+
+            // check: [left, right), error if left equal right
+            if (lowKey.compareTo(upperKey) == 0) {
```

Comment:
`lowKey.compareTo(upperKey) == 0` ==> `lowKey.compareTo(upperKey) >= 0` ?
If this is just for avoid `Range.closedOpen()`'s invalid check, you should add some comment here.

Reply:
ok